### PR TITLE
Update Ukrainian translation

### PR DESCRIPTION
**Pull request by AsturielUA**

### CODE REVIEW

File: files/lang/uk.po [hunk 1]
```diff
@@ -929,9 +902,8 @@ msgstr ""
 msgid "The mirror image is destroyed!"
 msgstr "Фантомний загін знищений!"
 
-#, fuzzy
 msgid "Are you sure you want to interrupt the auto battle?"
-msgstr "Ви впевнені, що хочете відступити?"
+msgstr "Ви впевнені, що хочете перервати "
```
[the remark this answers is not in the document]
Comment (AsturielUA) [reply files/lang/uk.po:906]: Done

File: files/lang/uk.po [hunk 2]
```diff
@@ -4154,9 +4125,8 @@ msgstr ""
 msgid "hotkey|toggle battle auto mode"
 msgstr ""
 
-#, fuzzy
 msgid "hotkey|finish the battle in auto mode"
-msgstr "Ви впевнені, що хочете відступити?"
+msgstr "Ви впевнені, що хочете завершити битву в автоматичному режимі?"
```
[the remark this answers is not in the document]
Comment (AsturielUA) [reply files/lang/uk.po:4148]: Done

File: files/lang/uk.po [hunk 3]
```diff
@@ -6680,69 +6649,53 @@ msgstr "Помаранчевий"
 msgid "Purple"
 msgstr "Фіолетовий"
 
-#, fuzzy
 msgid "barrier|Aqua"
-msgstr "Бірюзовий"
+msgstr "бірюзовий"
```
[the remark this answers is not in the document]
Comment (AsturielUA) [reply files/lang/uk.po:6691]: Program marked that wrong, asked to not use uppercase letters. I already realized how to change that, will change every string like that.

File: files/lang/uk.po [hunk 4]
```diff
@@ -7262,902 +7174,687 @@ msgstr "Вагонний табір"
...
 msgid_plural "Magic Gardens"
 msgstr[0] "Чарівний сад"
 msgstr[1] "Чарівні сади"
 msgstr[2] "Чарівних садів"
 
-#, fuzzy
 msgid "Observation Tower"
 msgstr "Оглядова вежа"
 
+# Name of object formed with two words. Straight translate sounds not very good: "Ливарня вільних людей". That phrase is not in use in ukrainian.
```
[the remark this answers is not in the document]
Comment (AsturielUA) [reply files/lang/uk.po:7382]: Will discuss it in next PR

File: files/lang/uk.po [hunk 5]
```diff
@@ -9777,22 +9463,22 @@ msgid "View Towns"
 msgstr ""
 
 msgid "Causes all Heroes across the land to become visible."
-msgstr ""
+msgstr "Показує всіх героїв на карті"
```
[the remark this answers is not in the document]
Comment (AsturielUA) [reply files/lang/uk.po:9504]: Done

File: files/lang/uk.po [hunk 6]
```diff
@@ -9777,22 +9463,22 @@ msgid "View Towns"
...
 
 msgid "View Heroes"
-msgstr ""
+msgstr "Показати Героїв"
 
 msgid "Causes the entire land to become visible."
-msgstr ""
+msgstr "Відкриває всю карту"
```
[the remark this answers is not in the document]
Comment (AsturielUA) [reply files/lang/uk.po:9510]: Done

File: files/lang/uk.po [hunk 7]
```diff
@@ -9860,59 +9546,65 @@ msgid "Sets Water Elementals to guard a mine against enemy armies."
...
+msgstr "Показати наступну сторінку"
 
 msgid "Close Spellbook"
-msgstr ""
+msgstr "Закрити Книгу Заклять"
 
 msgid "View %{spell}"
 msgstr ""
 
+#, fuzzy
```
[the remark this answers is not in the document]
Comment (AsturielUA) [reply files/lang/uk.po:9623]: I'm not sure I have understood that right.

[the remark this answers is not in the document]
Comment (archdron) [reply files/lang/uk.po:9623]: Screen
It's here

File: files/lang/uk.po [hunk 8]
```diff
@@ -9889,59 +9593,65 @@ msgid "Sets Water Elementals to guard a mine against enemy armies."
...
```
[the remark this answers is not in the document]
Comment (AsturielUA) [reply files/lang/uk.po:9640]: Before I thought that comments are concern string above, not below.)
Done.

File: files/lang/uk.po [hunk 9]
```diff
@@ -425,6 +422,7 @@ msgstr "Налаштувати"
 msgid "Hot Keys"
 msgstr "Гарячі клавіші"
 
+# Need to know where it used in game
```
[the remark this answers is not in the document]
Comment (AsturielUA) [reply files/lang/uk.po:425]: No, I can't see it in my poedit now

File: files/lang/Нова папка/uk.po [hunk 10]
```diff
@@ -0,0 +1,9730 @@
+# Ukrainian translation of fheroes2
```
[the remark this answers is not in the document]
Comment (AsturielUA) [reply files/lang/Нова папка/uk.po:1]: Program has found that backup folder. I've remored it away.

File: files/lang/uk.po (hunk 9 above)
[the remark this answers is not in the document]
Comment (AsturielUA) [reply files/lang/uk.po:425]: Found. Deleted.

**Pull request merged**